### PR TITLE
Cleanup patterns settings

### PR DESCRIPTION
**Pull request by thet**
Together:
https://github.com/plone/Products.CMFPlone/pull/3765
https://github.com/plone/plone.base/pull/33
https://github.com/plone/plone.api/pull/504
https://github.com/plone/plone.app.upgrade/pull/328

Jenkins:
https://jenkins.plone.org/job/pull-request-6.0-3.11/500

### CODE REVIEW

File: src/plone/base/interfaces/__init__.py [hunk 1]
```diff
@@ -26,7 +26,6 @@
 from .controlpanel import ITinyMCEPluginSchema
 from .controlpanel import ITinyMCEResourceTypesSchema
 from .controlpanel import ITinyMCESchema
-from .controlpanel import ITinyMCESpellCheckerSchema
```
Comment (jensens) [inline src/plone/base/interfaces/__init__.py:29]: Question: Do we need an upgrade step and remove the records from the registry?

Comment (petschki) [reply src/plone/base/interfaces/__init__.py:29]: yes 👍🏼

Comment (mauritsvanrees) [reply src/plone/base/interfaces/__init__.py:30]: @reinhardt is doing that in https://github.com/plone/plone.app.upgrade/pull/328
We are testing them all together now on Jenkins.

**Review by mauritsvanrees — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
I have rebased on main.

**Pull request merged**